src/zjs_i2c.c

```diff
@@ -110,25 +67,44 @@ static jerry_value_t zjs_i2c_read_base(const jerry_value_t this,
         return zjs_error("zjs_i2c_read_base: buffer creation failed");
     }
 
-    zjs_ipm_message_t send;
-    zjs_ipm_message_t reply;
-
     if (!burst) {
-        send.type = TYPE_I2C_READ;
-    } else {
-        send.type = TYPE_I2C_BURST_READ;
-    }
-
-    send.data.i2c.bus = (uint8_t)bus;
-    send.data.i2c.data = buf->buffer;
-    send.data.i2c.address = (uint16_t)address;
-    send.data.i2c.register_addr = register_addr;
-    send.data.i2c.length = size;
+        if (bus < MAX_I2C_BUS) {
+            // Read has to come after an Open I2C message
+            if (i2c_device[bus]) {
+
+                int reply = i2c_read(i2c_device[bus],
+                                     buf->buffer,
+                                     size,
+                                     (uint16_t)address);
+
+                if (reply < 0) {
+                    ZJS_PRINT("I2C Read failed with error : %i\n", reply);
```

Comment:
I'm not going to quibble more about this right now, not that important. But having a fresh think about it, this is definitely the kind of case where an exception is the right answer. Why should the read/write ever fail? There's no good reason, so it should be quite "exceptional" when it happens. Not something where we expect them to write a loop retrying because it may just fail any old time you call it.